tiledb/sm/serialization/array_schema.cc

```diff
@@ -1150,6 +1153,13 @@ shared_ptr<ArraySchema> array_schema_from_capnp(
     name = schema_reader.getName().cStr();
   }
 
+  auto crd = make_shared<CurrentDomain>(
```

Comment:
I think you'd like this to be in an `else` case, otherwise if there's a serialized domain you construct twice and just throw away the first one.
But IMO this "empty"/default object construction should be happening inside the `ArraySchema` constructor itself, if the caller is passing an uninitialized `shared_ptr` (aka `nullptr`). Otherwise we leave the door open to someone not knowing about this and just passing a `nullptr` for `current_domain` when constructing an `ArraySchema` opening the door to segfaults.. (didn't see any checks for null `current_domain` in `ArraySchema` class)

Reply:
This is an easy change, I adjusted the code. In general, I'm shooting for code simplicity in this area, unless I'm doing something with disk access that could be in the same order of magnitude with the wire access, I'm not even thinking of performance.
About the empty current domain construction, there is another constructor for `ArraySchema` that does just that, this particular one is used [only]? in serialization and assumes we pass the correct things, the argument with passing nullptr by mistake can be extended to all other `shared_ptr` args we pass to this constructor.